### PR TITLE
Implementations of "linear" objects for optimal control

### PR DESCRIPTION
**Pull request by S-Dafarra**
Implements few classes to ease the definition of linear optimal control problems. For the time being, only ``OSQP`` is wrapped (through [``osqp-eigen``](https://github.com/robotology/osqp-eigen)).

### CODE REVIEW

File: src/optimalcontrol/CMakeLists.txt [hunk 1]
```diff
@@ -69,6 +72,22 @@ if (IDYNTREE_USES_IPOPT)
     list(APPEND INCLUDE_LIST ${IPOPT_INCLUDE_DIRS})
 endif()
 
+find_package(OsqpEigen QUIET)
```
Comment (traversaro) [inline src/optimalcontrol/CMakeLists.txt]: Can you add a `idyntree_handle_dependency(OsqpEigen)` line in https://github.com/robotology/idyntree/blob/devel/cmake/iDynTreeDependencies.cmake#L56 and use the `IDYNTREE_USES_OSQPEIGEN` to select if the dependency is used or not? Thanks!

**Comment by S-Dafarra** [comment]
The ``iDynTreeWholeBodyFTsensorEstimate`` test failed in one of the Travis jobs :thinking:.

**Comment by traversaro** [comment]
If @GiulioRomualdi eventually reviews the code, I am ok in merging with master to avoid branch fragmentation. I would just ask @S-Dafarra to add a line w.r.t. to this changes in https://github.com/robotology/idyntree/blob/devel/doc/releases/v0_11.md .

**Comment by S-Dafarra** [comment]
Ok I will wait for some reviews before changing the release notes to avoid queuing travis jobs only for a text document.

**Comment by traversaro** [comment]
> Ok I will wait for some reviews before changing the release notes to avoid queuing travis jobs only for a text document.

If you in the commit description you include the line `[ci skip]`  or `[skip ci]`, no travis job is created (and that usually make sense for doc-only modifications.

**Comment by traversaro** [comment]
@S-Dafarra do you think someone can check this this week? I would like release 0.11 this week.

**Comment by S-Dafarra** [comment]
I actually don't know. I would be happier to have some feedback and wait for the next release rather than hurry and having problems later on.

**Pull request merged**